sdk/core/Azure.Core/src/Shared/OperationInternalBase.cs

```diff
@@ -198,43 +236,37 @@ private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToke
 
             try
             {
-                return await UpdateStateAsync(async, cancellationToken).ConfigureAwait(false);
-            }
-            catch (Exception e)
-            {
-                scope.Failed(e);
-                throw;
-            }
-        }
-
-        protected async ValueTask<Response> ApplyStateAsync(bool async, Response response, bool hasCompleted, bool hasSucceeded, RequestFailedException? requestFailedException, bool throwIfFailed = true)
-        {
-            RawResponse = response;
+                var state = await UpdateStateAsync(async, cancellationToken).ConfigureAwait(false);
+                RawResponse = state.RawResponse;
+                if (!state.HasCompleted)
+                {
+                    return state.RawResponse;
+                }
 
-            if (!hasCompleted)
-            {
-                return response;
-            }
+                if (!state.HasSucceeded && state.OperationFailedException == null)
+                {
+                    var exception = async
+                        ? await _diagnostics.CreateRequestFailedExceptionAsync(state.RawResponse).ConfigureAwait(false)
```

Comment:
@annelo-msft when custom error parsing introduced, we will have to update this place as well